### PR TITLE
DDF-2315: Add support for download cancellation to the CometDClient

### PR DESCRIPTION
**Pull request by figliold**
#### What does this PR do?
Adds support for download cancellation to the CometDClient used in itests
@bantillo 
@garrettfreibott 
@Lambeaux 
@oconnormi 
@cmthom10 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@shaundmorris 
#### How should this be tested?
Hero build
#### Any background context you want to provide?
PR https://github.com/codice/ddf/pull/1016 needs to be merged before this PR can be merged
#### What are the relevant tickets?
DDF-2315
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [x] Update / Add Integration Tests

### CODE REVIEW

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java [hunk 1]
```diff
@@ -1474,6 +1538,65 @@ public void testProductDownloadListWithOneActiveDownload() {
...
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(200), 0));
+
+        String startDownloadUrl = RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl() + "?source="
+                + CSW_STUB_SOURCE_ID + "&metacard=" + metacardId;
+
```
Comment (cmthom10) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: In my PR (https://github.com/codice/ddf/pull/1009/files) I created a DownloadHandle class that could probably be used here as well to set up lines 1551 through 1565.

**Comment by coyotesqrl** [comment]
Conflicts to be resolved.

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/AbstractEventController.java [hunk 2]
```diff
@@ -230,7 +230,9 @@ public void registerUserSession(final ServerSession serverSession, ServerMessage
 
         userSessionMap.put(userId, serverSession);
 
-        LOGGER.debug("Added ServerSession to userSessionMap - New map: {}", userSessionMap);
+        LOGGER.debug("Added ServerSession for user {} to userSessionMap - New map: {}",
```
Comment (shaundmorris) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/AbstractEventController.java]: @figliold should be fine. @oconnormi  made the same formatting changes in https://github.com/codice/ddf/pull/1027

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/NotificationController.java [hunk 3]
```diff
@@ -81,41 +81,52 @@ public void handleEvent(Event event) throws IllegalArgumentException {
                 || event.getProperty(Notification.NOTIFICATION_KEY_APPLICATION)
                 .toString()
                 .isEmpty()) {
-            throw new IllegalArgumentException(
-                    "Event \"" + Notification.NOTIFICATION_KEY_APPLICATION
-                            + "\" property is null or empty");
+            String message = "Event [" + Notification.NOTIFICATION_KEY_APPLICATION
```
Comment (shaundmorris) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/NotificationController.java]: @figliold should work with @oconnormi on how and who will resolve using `String.format()` which was a comment on https://github.com/codice/ddf/pull/1027

**Comment by shaundmorris** [comment]
Should we merge https://github.com/codice/ddf/pull/1027 and rebase this PR? I see you have @oconnormi's commit on this PR as well.

File: catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/NotificationController.java (hunk 3 above)
Comment (figliold) [inline catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/NotificationController.java]: fixed in PR https://github.com/codice/ddf/pull/1016

File: distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java (hunk 1 above)
Comment (figliold) [inline distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java]: That is a great idea.  I think I will write a ticket to make use of the DownloadHandle code in all of our download itests and put in next sprint's backlog.

**Comment by figliold** [comment]
Conflicts have been resolved.

**Comment by figliold** [comment]
This ticket requires https://github.com/codice/ddf/pull/1016 which has already been merged.

**Comment by oconnormi** [comment]
LGTM 👍 
Just make sure that we use the second commit message as the actual one when this get's squashed ;-)

**Comment by bantillo** [comment]
👍 , same LGTM

**Comment by coyotesqrl** [comment]
LGTM

**Comment by shaundmorris** [comment]
👍

**Comment by Lambeaux** [comment]
👍

**Comment by cmthom10** [comment]
Successfully hero'd!

**Pull request merged**